Extended tab auto complete to work with bundler and custom commands

revolter opened the pull request:
<!-- Thanks for contributing to _fastlane_! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] -->

### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [ ] I've updated the documentation if necessary.

### Motivation and Context
<!--- Why is this change required? What problem does it solve? -->

This enabled tab auto complete for `bundle exec fastlane` and allows passing custom commands like `fl` to enable tab auto complete for `./fl`.

<!--- If it fixes an open issue, please link to the issue here. -->
<!--- Please describe in detail how you tested your changes. --->

Works with:

    bundle exec fastlane enable_auto_complete
    bundle exec fastlane enable_auto_complete --custom first
    bundle exec fastlane enable_auto_complete --custom first,second

and:

    sh('bundle exec fastlane enable_auto_complete --custom first')

in a lane in the `Fastfile`.

### Description
<!--- Describe your changes in detail -->

revolter committed:
Enable auto complete when using bundler

This enables typing "bundle exec fastlane lane_nam" and pressing
tab to auto complete with "lane_name".

The trade-off is that it also works when typing
"bundle exec lane_nam" or even "bundle lane_nam", but I think it's
an acceptable trade-off, as the chances of something thinking it
can work this way seems low.

revolter committed:
Allow enable_auto_complete command to be called non-interactively

revolter committed:
Add option to enable_auto_complete to specify custom commands

Tab auto complete will work for the specified custom commands as
it does for "fastlane" and "bundle exec fastlane".

You use it by passing them to the "custom" option like:
bundle exec fastlane enable_auto_complete --custom fl,bundle_fastlane

milch commented on fastlane/lib/fastlane/commands_generator.rb:
```diff
@@ -218,10 +218,11 @@ def run
       command :enable_auto_complete do |c|
         c.syntax = 'fastlane enable_auto_complete'
         c.description = 'Enable tab auto completion'
+        c.option '-c STRING[,STRING2]', '--custom STRING[,STRING2]', String, 'Add custom command(s) to use for auto complete'
```
Could you expand a bit on this comment? From this description it sounds a bit unclear to me what this does

revolter replied:
> Add custom command(s) for which tab auto complete should work too

milch replied:
Yup, sounds better to me! Maybe `should be enabled too` instead of `should work too`

milch reviewed (commented):
I like it, although I do think the possibility for it completing `bundle lane_name` could be a bit of an issue. Do we think that only advanced users will install shell completion anyway, so these users would not be confused by this? It looks like the auto completion for bundle is also added when calling `fastlane enable_auto_complete` without the `bundle exec`. Maybe we could work around any confusion by only adding the completion if it is called with `bundle exec`?

revolter commented:
Indeed, installing completion generally means that you are accustomed to the terminal, and if you need to use the `bundle` command, chances that you know how it works are pretty high. In the worst case scenarios, `bundler` would print `Could not find command "deploy".` and `bundler: command not found: deploy`. Having no auto complete at all seems a worse solution, and there doesn't seem to be any way for it to work when typing the entire `bundle exec fastlane` string.

revolter committed:
Improve custom option description message

revolter committed:
Replace duplicate code with enum

revolter committed:
Remove superfluous auto complete lines

This enables auto complete for "fastlane" if running
"enable_auto_complete" directly with fastlane and for "bundle exec
fastlane" if running via bundler

revolter committed:
Fix code style

revolter committed:
Fix missing default auto complete line

- introduced by 2ab0ffbc
